src/main/kotlin/org/web3j/strategy/CustomWaitingStrategy.kt

```diff
@@ -0,0 +1,285 @@
+/*
+ * Copyright 2020 Web3 Labs Ltd.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+package org.web3j.strategy
+
+import org.rnorth.ducttape.TimeoutException
+import org.rnorth.ducttape.unreliables.Unreliables
+import org.testcontainers.containers.ContainerLaunchException
+import org.testcontainers.containers.wait.strategy.AbstractWaitStrategy
+import org.testcontainers.shaded.com.google.common.base.Strings
+import org.testcontainers.shaded.com.google.common.io.BaseEncoding
+import java.io.BufferedReader
+import java.io.IOException
+import java.io.InputStreamReader
+import java.net.HttpURLConnection
+import java.net.URI
+import java.net.URL
+import java.time.Duration
+import java.util.Optional
+import java.util.concurrent.TimeUnit
+import java.util.function.Predicate
+import kotlin.collections.HashSet
+
+class CustomWaitingStrategy : AbstractWaitStrategy() {
```

Comment:
Why not inherit from `HttpWaitStrategy`? It will save you a few lines of code.

Reply:
Also the naming is very general. Something like `HttpMethodWaitStrategy` or so would make the intent of the class clearer. 

Otherwise you should put some Javadoc why this class is needed, is because `HttpWaitStrategy` doesn't allow defining the HTTP method.